File: app/src/main/java/com/jakebarnby/filemanager/sources/models/SourceManager.java
package com.jakebarnby.filemanager.sources.models;

import android.util.SparseArray;

import com.jakebarnby.filemanager.models.FileAction;
import com.jakebarnby.filemanager.sources.SourceActivity;
import com.jakebarnby.filemanager.util.TreeNode;

/**
 * Created by Jake on 7/29/2017.
 */

public class SourceManager {

    private TreeNode<SourceFile>                    mActiveDirectory;
    private SparseArray<FileAction>  mCurrentFileActions;

    public SourceManager() {
        this.mCurrentFileActions = new SparseArray<>();
    }

    public TreeNode<SourceFile> getActiveDirectory() {
        return mActiveDirectory;
    }

    public void setActiveDirectory(TreeNode<SourceFile> mActiveDirectory) {
        this.mActiveDirectory = mActiveDirectory;
    }

    public FileAction getFileAction(int operationId) {
        return mCurrentFileActions.get(operationId);
    }

    public void addFileAction(int operationId, FileAction action) {
        mCurrentFileActions.put(operationId, action);
    }
}

File: app/src/main/java/com/jakebarnby/filemanager/sources/models/SourceManager.kt
package com.jakebarnby.filemanager.sources.models

import android.util.SparseArray
import com.jakebarnby.filemanager.models.FileAction
import java.io.Serializable

/**
 * Created by Jake on 7/29/2017.
 */
class SourceManager(
    val sources: MutableList<Source> = mutableListOf()
) : Serializable {

    lateinit var activeSource: Source

    private val currentFileActions = SparseArray<FileAction>()

    fun getFileAction(operationId: Int): FileAction? {
        return currentFileActions[operationId]
    }

    fun addFileAction(operationId: Int, action: FileAction) {
        currentFileActions.put(operationId, action)
    }

}